src/actions/gifs.js

```diff
@@ -0,0 +1,604 @@
+
+        const extendedParams = {
+            gfyid: gfyId,
+            ...params,
+            device_type: deviceType,
+            utc: encodedUtc,
+            stc: encodedStc,
+            rand: Math.random() * 100000,
+        };
+
+        const queryString = queryParamsString(extendedParams);
+        const url = `https://px.gfycat.com/px.gif?${queryString}`;
+        fetch(url);
+    };
+}
+
+export function sendViewCountBatch({gfycats, params}) {
+    return (dispatch, getState) => {
+        if (typeof document === 'undefined') {
+            return;
+        }
+        if (!params.app_id) {
+            params.add_id = 'com.gfycat.website';
+        }
+
+        const {utc, stc} = getUserAndSessionIds({dispatch, getState});
+        const encodedUtc = encodeURIComponent(utc);
+        const encodedStc = encodeURIComponent(stc);
+
+        if (gfycats.length > BATCH_MAX_LENGTH) {
+            const chunks = [];
+            while (gfycats.length) {
+                chunks.push(gfycats.splice(0, BATCH_MAX_LENGTH));
+            }
+            for (let i = 0; i < chunks.length; i++) {
+                dispatch(sendViewCountBatch({gfycats: chunks[i], params}));
+            }
+            return;
+        }
+
+        const gfyIds = gfycats.reduce((ids, gfyItem, index) => {
+            const key = index === 0 ? 'gfyid' : `gfyid_${index}`;
+            ids[key] = gfyItem.gfyId;
+            return ids;
+        }, {});
+
+        const extendedParams = {
+            ...params,
+            ...gfyIds,
+            utc: encodedUtc,
+            stc: encodedStc,
+        };
+
+        const queryString = queryParamsString(extendedParams);
+
+        const url = `https://px.gfycat.com/px.gif?${queryString}`;
+        fetch(url);
+    };
+}
+
+export function splitBatchRequest({gfycats, params, dispatch}) {
+    dispatch(sendViewCountBatch({gfycats, params}));
+}
+
+export function sendEvent({event, params}) {
+    return (dispatch, getState) => {
+        if (typeof document === 'undefined') {
+            return;
+        }
+        dispatch(analyticsEvent({event, params}));
+        const {utc, stc} = getUserAndSessionIds({dispatch, getState});
+        const encodedUtc = encodeURIComponent(utc);
+        const encodedStc = encodeURIComponent(stc);
+        const ref = typeof document.referrer !== 'undefined' && document.referrer.length ?
+            document.referrer : 'https://gfycat.com';
+        const deviceType = 'unknown';
+
+        const extendedParams = {
+            ...params,
+            event,
+            utc: encodedUtc,
+            stc: encodedStc,
+            ref,
+            device_type: deviceType,
+        };
+
+        const queryString = queryParamsString(extendedParams);
+
+        const url = `https://metrics.gfycat.com/pix.gif?${queryString}`;
+        fetch(url);
+    };
+}
+
+function queryParamsString(params) {
+    return params ? Object.keys(params).map((key) => {
```

Comment:
Can this function be refactored like
```
if (!params) {
    return '';
}

return Object.keys(params)...;
```
to follow our conventions?

Reply:
Done.